dbt/adapters/duckdb/environments.py

```diff
@@ -87,16 +85,13 @@ def cursor(self):
             cursor.execute(f"SET {key} = '{value}'")
         return cursor
 
-    def close(self, cursor):
-        cursor.close()
-        self.handles -= 1
-        if self.conn and self.handles == 0 and self.creds.path != ":memory:":
+    def close(self):
+        if self.conn:
```

Comment:
So I thought i needed this handling logic for the in-memory stuff to work correctly, but maybe I'm wrong?

Reply:
If you don't take the `self.creds.path != ":memory:"` part into account, the handling logic produces this behaviour:
1. Create connection to duckdb -- handles = 0
2. Create cursor -- handles = 1
3. Release cursor -- handles = 0
4. Release connection because handles == 0
5. Create a new cursor, but throw an error because the duckdb connection is already closed.

By putting the extra `:memory`: check, closing the connection prematurely is prevented, but the problem will then still present itself when running against a file-backed db. Skipping the check for both memory and file seems a bit redundant, so I just removed it.

I am running the tests against dbt 1.4.5 and duckdb 0.7.1, maybe the behavior was a bit different on older versions?

Reply:
yeah, I'm remembering why I did this now-- https://getdbt.slack.com/archives/C039D1J1LA2/p1665015877499679

The context there is that if you're running fal alongside dbt-duckdb, and one of the fal scripts needs to use the DuckDB connection, it can't access it b/c the dbt-duckdb process owns the write lock on the file as long as one of the connections is open. So the point of this tracking was to have the opportunity for dbt-duckdb to release the write lock completely whenever the last connection to the DB was closed so that the fal script would have the opportunity to take it over.

In the new world of dbt-fal, this may be less relevant and thus no longer worth keeping around, but I'd hate to break anybody who was running this as-is, so I'll solicit some feedback in the dbt Slack to see if taking it out is an issue for folks; there are some pretty obvious code clarity/perf wins to not having to open and close the connection over and over again.